File: app/src/main/java/com/boardgamegeek/provider/GamesIdPublishersIdProvider.java
package com.boardgamegeek.provider;

import android.content.ContentUris;
import android.net.Uri;

import com.boardgamegeek.provider.BggContract.Publishers;
import com.boardgamegeek.provider.BggDatabase.GamesPublishers;
import com.boardgamegeek.util.SelectionBuilder;

public class GamesIdPublishersIdProvider extends BaseProvider {
	final GamesIdPublishersProvider provider = new GamesIdPublishersProvider();

	@Override
	protected SelectionBuilder buildSimpleSelection(Uri uri) {
		final long publisherId = ContentUris.parseId(uri);
		return provider.buildSimpleSelection(uri).whereEquals(GamesPublishers.PUBLISHER_ID, publisherId);
	}

	@Override
	protected String getPath() {
		return addIdToPath(provider.getPath());
	}

	@Override
	protected String getType(Uri uri) {
		return Publishers.CONTENT_ITEM_TYPE;
	}
}

File: app/src/main/java/com/boardgamegeek/provider/GamesIdPublishersIdProvider.kt
package com.boardgamegeek.provider

import android.content.ContentUris
import android.net.Uri
import com.boardgamegeek.provider.BggContract.*
import com.boardgamegeek.provider.BggContract.Companion.PATH_GAMES
import com.boardgamegeek.provider.BggContract.Companion.PATH_PUBLISHERS
import com.boardgamegeek.provider.BggDatabase.GamesPublishers.GAME_ID
import com.boardgamegeek.provider.BggDatabase.GamesPublishers.PUBLISHER_ID
import com.boardgamegeek.provider.BggDatabase.Tables

class GamesIdPublishersIdProvider : BaseProvider() {
    override fun getType(uri: Uri) = Publishers.CONTENT_ITEM_TYPE

    override val path = "$PATH_GAMES/#/$PATH_PUBLISHERS/#"

    override fun buildSimpleSelection(uri: Uri): SelectionBuilder {
        val gameId = Games.getGameId(uri)
        val publisherId = ContentUris.parseId(uri)
        return SelectionBuilder()
            .table(Tables.GAMES_PUBLISHERS)
            .whereEquals(GAME_ID, gameId)
            .whereEquals(PUBLISHER_ID, publisherId)
    }
}
